### PR TITLE
Prevent missile launcher ammo unlocking without allowGuidedLaunchers

### LINKED ISSUE #509: Guided missile ammo unlocks (easily) regardless of the launcher param
Version: Unstable.

All magazines are currently unlocked at the base minWeaps count, including those for guided missile launchers (eg. Stinger, Javelin, Titan). This is probably unintentional, and makes it easy for players to defeat air attacks as they can load up a car with a dozen missiles.

Suggested fix: Detect if magazines contain missiles. If so, don't allow them to be unlocked unless the unlockable missile launcher param is enabled.

### PR DESCRIPTION
**Pull request by jaj22**
## What type of PR is this.
1. [X] Bug
2. [ ] Enhancement

### What have you changed and why?
On unstable, missile launcher ammo currently unlocks at minWeaps, regardless of whether missile launcher unlocking is allowed. This code prevents missile ammunition unlocking unless `allowGuidedLaunchers` is true.

I retained the style of declaring the private variables outside the loop, but the performance gain is very small (checked with diag_codeperformance) even for trivial loops, so I wouldn't recommend it here.

One oddity is that this code blocks PCML ammunition from unlocking, because the PCML is classified as a rocket launcher by BIS_fnc_itemType, but the ammunition is a missile. As it's probably a BIS bug, it could be worked around with a `_type == "NLAW_F"` check if necessary, rather than grinding through all the launchers and categorising their ammo accordingly.

### Please specify which Issue this PR Resolves.
closes #509 

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the Mission in Singleplayer?
2. [X] Have you loaded the Mission in a Dedicated Server?
Not relevant but I did it anyway.

### Is further testing or are further changes required?
1. [X] No, Unless you want the PCML hack.
2. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

File: A3-Antistasi/functions/Ammunition/fn_arsenalManage.sqf [hunk 1]
```diff
@@ -18,14 +18,22 @@ private _vests = (jna_dataList select IDC_RSCDISPLAYARSENAL_TAB_VEST) select {_x
...
 	_type = _x select 0;
-	_capacity = getNumber (configFile >> "CfgMagazines" >> _type >> "count");
-	_bullets = _x select 1;
-	_count = floor (_bullets/_capacity);
-	_magazine pushBack [_type,_count];
+	_magConfig = configFile >> "CfgMagazines" >> _type;
+	_capacity = getNumber (_magConfig >> "count");
+
+	// control unlocking missile launcher magazines
+	if (_capacity != 1 || allowGuidedLaunchers isEqualTo 1 ||
```
Comment (Spoffy) [inline A3-Antistasi/functions/Ammunition/fn_arsenalManage.sqf]: Do we need the capacity check in here? I don't think it'll ever actually do anything, will it? Or is there some edge case here I'm missing?

**Comment by jaj22** [comment]
It's a theoretical lazy-execution performance optimisation for the case where you're not running allowGuidedLaunchers, so that it doesn't bother with the config lookup unless it's a plausible candidate for being a missile magazine. The assumption is that config lookups are relatively expensive, which I haven't checked.

**Comment by Spoffy** [comment]
They are quite expensive, if memory serves, but I couldn't put a number on it. Alright, I'm happy with that :) - can you put it in as a comment, so whoever ends up reading this knows why it's there, and exactly how important it is :)
:)

**Comment by jaj22** [comment]
Some additional detail: On a mid-game vanilla arsenal (which contains a surprising amount of random garbage now, so loop execution was around 0.5 secs), that optimisation is about twice as valuable as declaring the private vars outside the loop, and very close to replacing with `if (true || ...`.

**Pull request merged**